crypto/bio/bio_mem.c

```diff
@@ -231,6 +294,9 @@ static long mem_ctrl(BIO *bio, int cmd, long num, void *ptr) {
     case BIO_CTRL_INFO:
       ret = (long)b->length;
       if (ptr != NULL) {
+        if (!(bio->flags & BIO_FLAGS_MEM_RDONLY)) {
+          mem_buf_sync(bio);
```

Comment:
why do we `mem_buf_sync` here? OpenSSL [doesn't](https://github.com/openssl/openssl/blob/995e9489e62ff1965553c4e127183565ccfe4265/crypto/bio/bss_mem.c#L294).

Reply:
Good catch! Glossed over this part for OpenSSL